lib/proxy/http_download.rb

```diff
@@ -5,11 +5,8 @@ class HttpDownload < Proxy::Util::CommandTask
     include Util
     DEFAULT_CONNECT_TIMEOUT = 10
 
-    def initialize(src, dst, read_timeout = nil, connect_timeout = nil, dns_timeout = nil, verify_server_cert = false)
+    def initialize(src, dst, connect_timeout: DEFAULT_CONNECT_TIMEOUT, verify_server_cert: false)
       @dst = dst
-      logger.warn('Deprecated: HttpDownload read_timeout is deprecated and will be removed in 4.0') if read_timeout
-      logger.warn('Deprecated: HttpDownload dns_timeout is deprecated and will be removed in 4.0') if dns_timeout
-      connect_timeout ||= DEFAULT_CONNECT_TIMEOUT
```

Comment:
technically, we're not releasing 4.0 right now, so we can't remove that here?

Reply:
Hmm, good question. Should we change it to "removed in 3.5" in https://github.com/theforeman/smart-proxy/pull/841 instead?

Reply:
I'd be cool with that.

Reply:
Done and merged.